pkg/resources/server.go

```diff
@@ -201,57 +215,53 @@ func (m *resourceServer) Restart() error {
 }
 
 // Watch watch for changes in old socket if used
-func (m *resourceServer) Watch() {
+func (m *resourceServer) Watch() error {
 	log.Println("Starting FS watcher.")
 	watcher, err := newFSWatcher(deprecatedSockDir)
 	if err != nil {
-		log.Println("Failed to created FS watcher.")
-		os.Exit(1)
+		log.Fatal("Failed to created FS watcher.")
 	}
+	m.socketWatcher = watcher
 	defer watcher.Close()
 
 	select {
 	case event := <-watcher.Events:
 		if event.Name == m.socketPath && event.Op&fsnotify.Create == fsnotify.Create {
 			log.Printf("inotify: %s created, restarting.", m.socketPath)
 			if err = m.Restart(); err != nil {
-				log.Fatalf("unable to restart server %v", err)
+				return fmt.Errorf("unable to restart server %v", err)
 			}
 		}
 
 	case err := <-watcher.Errors:
-		log.Printf("inotify: %s", err)
+		return fmt.Errorf("inotify: %s", err)
 
 	case stop := <-m.stopWatcher:
 		if stop {
 			log.Println("kubelet watcher stopped")
-			watcher.Close()
-			return
+			_ = watcher.Close()
```

Comment:
no need to check retval ?

Reply:
add a check for it

Reply:
not directly related to this patch, but should it be checked ?

Reply:
ok, maybe in another patch we can add check for all not checked closed, there are others also

Reply:
great, thanks!